icepyx/core/validate_inputs.py

```diff
@@ -29,70 +29,94 @@ def dset_version(latest_vers, version):
     return vers
 
 
-def cycles(all_cycles, cycles):
+def cycles(cycle):
     """
     Check if the submitted cycle is valid, and warn the user if not available.
     """
-    if cycles is None:
-        cycle_list = all_cycles
+    cycle_length = 2
+    # number of GPS seconds between the GPS epoch and ATLAS SDP epoch
+    atlas_sdp_gps_epoch = 1198800018.0
+    # number of GPS seconds since the GPS epoch for first ATLAS data point
+    atlas_gps_start_time = atlas_sdp_gps_epoch + 24710205.39202261
+    epoch1 = dt.datetime(1980,1,6,0,0,0)
+    epoch2 = dt.datetime(1970,1,1,0,0,0)
+    # get the total number of seconds since the start of ATLAS and now
+    delta_time_epochs = (epoch2 - epoch1).total_seconds()
+    atlas_UNIX_start_time = (atlas_gps_start_time - delta_time_epochs)
+    present_time = dt.datetime.now().timestamp()
+    # divide total time by cycle length to get the maximum number of orbital cycles
+    ncycles = np.ceil((present_time - atlas_UNIX_start_time)/(86400*91)).astype('i')
+    all_cycles = [str(c+1).zfill(cycle_length) for c in range(ncycles)]
+
+    if cycle is None:
+        return all_cycles
```

Comment:
What is the incentive for returning `all_cycles` here versus an empty list? If the user doesn't specify cycles (or tracks), then this function wouldn't be called, and `region_a.cycles` doesn't get set. Currently, if no time is set during initialization then the property (`region_a.dates`) is unavailable (I just made that change to prevent errors). I think we'll decrease confusion if we keep the behavior consistent across unused parameters. Compelling arguments for one route or the other encouraged. : )

Reply:
Good point!  It simplified making the `readable_granule_name` list but you're right.  I'll put in a change to just add the appropriate wildcard if either cycles or tracks is excluded.